### PR TITLE
storeapi: add support for reporting status of progressive releases

### PR DESCRIPTION
**Pull request by maxiberta**
Parse the [`"current-percentage"` Snap Store API field](https://dashboard.staging.snapcraft.io/docs/api/snap.html#about-progressive-releases) and show the current status of progressive releases on CLI.

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [ ] Have you successfully run `./runtests.sh static`?
  - `shellcheck` throws a bunch of warnings, but I believe these are not related with this at all.
- [x] Have you successfully run `./runtests.sh tests/unit`?

-----

### CODE REVIEW

**Comment by maxiberta** [comment]
Sample output (on staging):

```bash
snapcraft staging servers> SNAPCRAFT_EXPERIMENTAL_PROGRESSIVE_RELEASES=y snapcraft status htop-snap
*EXPERIMENTAL* progressive releases in use.
Track    Arch    Channel    Version            Revision    Progress
latest   amd64   stable     2.2.0-20-g402e46b  148         85 → 80%
                            2.2.0-20-g402e46b  149         15 → 20%
                 candidate  2.2.0-20-g402e46b  149         -
                 beta       2.2.0-20-g402e46b  150         -
                 edge       2.2.0-20-g402e46b  154         100 → 65%
                            2.2.0-20-g402e46b  155         0 → 35%
```

**Comment by maxiberta** [comment]
As in [my previous PR](https://github.com/snapcore/snapcraft/pull/3305), `tests/spread/plugins/v1/flutter/run` is failing with `flutter: "build linux" is not currently supported`.

**Comment by sergiusens** [comment]
No worries about the flutter test, that is being investigated.

**Comment by sergiusens** [comment]
Thank you for you contribution to Snapcraft, I will be looking into this shortly

File: snapcraft/storeapi/v2/_api_schema.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
                         "properties": {
                             "paused": {"type": ["boolean", "null"]},
                             "percentage": {"type": ["number", "null"]},
+                            "current-percentage": {"type": ["number", "null"]},
                         },
                         "required": ["paused", "percentage"],
```
Comment (cjp256) [inline snapcraft/storeapi/v2/_api_schema.py:46]: i think 'current-percentage' a required field now?

Comment (maxiberta) [reply snapcraft/storeapi/v2/_api_schema.py]: Good catch! Fixed. Thanks

File: snapcraft/cli/_channel_map.py [hunk 2]
```diff
@@ -123,20 +123,28 @@ def _get_channel_lines_for_channel(  # noqa: C901
             progressive_mapped_channel.revision
         )
 
+        if progressive_mapped_channel.progressive.percentage is None:
```
Comment (sergiusens) [inline snapcraft/cli/_channel_map.py:126]: this should probably raise a RuntimeError as "get_mapped_channel" with progressive True will return a channel where percentage is set

Comment (maxiberta) [reply snapcraft/cli/_channel_map.py:127]: Makes sense! Updated.

**Review by sergiusens — APPROVED** [review]
Thank you for your contribution!

**Comment by sergiusens** [comment]
The failure is unrelated (electron builder failure)

**Pull request merged**